packages/client/src/workspace/home/index.ts

```diff
@@ -85,22 +86,34 @@ export async function registerHome(): Promise<HomeRegistration> {
     return getAppsAndPages(query.length < queryMinLength ? undefined : query)
   }
 
-  const handlePageSelection = async (result: CLIDispatchedSearchResult) => {
+  type PageSearchResult = Omit<CLIDispatchedSearchResult, "data"> & {
+    data: PageData
+  }
+
+  const handlePageSelection = async (result: PageSearchResult) => {
     if (result.action.name === HOME_ACTION_DELETE_PAGE) {
       await deletePage(result.data.pageId)
       if (lastResponse !== undefined && lastResponse !== null) {
         lastResponse.revoke(result.key)
       }
     } else {
       const pageToLaunch = await getPage(result.data.pageId)
-      await launchPage(pageToLaunch)
+      if (pageToLaunch) {
+        await launchPage(pageToLaunch)
+      } else {
+        console.warn(`No page found for ${result.data.pageId}`)
+      }
     }
   }
 
+  const isPageData = (
```

Comment:
bit of "long hand" typing .. to tie the page data with its source in `utils.ts` and give a narrowed type for the page search result, as it is totally different from app/view result, at the "data" level